src/components/icon/_icon.scss

```
@@ -40,14 +40,13 @@
   }
 }
 
-// App icon special cases for fully mono
-.euiIcon--text,
-.euiIcon--subdued,
-.euiIcon--primary,
-.euiIcon--customColor {
-  &,
-  .euiIcon__fillSecondary {
-    fill: currentColor;
+// Really force all paths to inherit when the color is ghost/text (specifically for logos)
+.euiIcon--ghost:not([data-type='logoElastic']),
+.euiIcon--text:not([data-type='logoElastic']) {
+  > *,
+  > * > *,
+  > * > * > * {
```

This way we can can match any element:
 
```suggestion
  * {
```

---

Haha, yeah that's probably more efficient. But then you got me thinking about it possibly targeting def's or other non-shape/path elements, and so I'm wondering if we should target just those with a `fill` attribute. So like `*[fill]`. I'm going to try that locally and see how it works.

---

Love the selector simplification! 👏